### ISSUE TITLE
Valiant Hearts: The Great War (260230)

### ISSUE THREAD

**Issue by perroboc** [ValveSoftware/Proton#6811]
# Compatibility Report
- Name of the game with compatibility issues: Valiant Hearts: The Great War
- Steam AppID of the game: 260230

## System Information
- GPU: RX 7900 XTX
- Driver/LLVM version: Mesa 23.1.1
- Kernel version: 6.3.4-arch1-1
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/alvaromunoz/ae7c9dec0f5a87de1da27831443451ed#file-sysinfo-txt
- Proton version: 8.0-2, Experimental, Experimental bleeding edge

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

## Symptoms 
UPlay, or Ubisoft Connect, is not able to connect to the Ubisoft servers, so it crashes with this message:
![image](https://github.com/ValveSoftware/Proton/assets/2722773/c632eded-5e01-4da3-9e74-ee1c116d6d85)

## Reproduction
Launch the game and login to UPlay.

I've also installed the latest Ubisoft Connect in the game's wineprefix, but since that version tries to connect before login, it immediatly crashes with the same message.

**Comment by perroboc** [comment]
proton experimental log: 
[steam-260230.log](https://github.com/ValveSoftware/Proton/files/11614604/steam-260230.log)

proton experimental bleeding-edge-debug log: 
[steam-260230-bed.log](https://github.com/ValveSoftware/Proton/files/11614664/steam-260230-bed.log)

**Comment by kisak-valve** [comment]
Hello @alvaromunoz, `warn:debugstr:OutputDebugStringA "C:\\Jenkins_p4ws\\UPLAY-PC-STREAMS\\client\\release_baldur_patch3-live\\shared\\libs\\common\\mg/common/Json.h(295) : ASSERT: Wrong value type\n"` looks like a possible line of interest from your log.

**Comment by perroboc** [comment]
> Hello @alvaromunoz, `warn:debugstr:OutputDebugStringA "C:\\Jenkins_p4ws\\UPLAY-PC-STREAMS\\client\\release_baldur_patch3-live\\shared\\libs\\common\\mg/common/Json.h(295) : ASSERT: Wrong value type\n"` looks like a possible line of interest from your log.

Yeah, I was actually looking at that from a previous post from you https://github.com/ValveSoftware/Proton/issues/2425#issuecomment-1475290067 but it seems that Jenkins path is the codename of the Ubisoft launcher, and the old versions had nordic codenames (baldur, heimdall). Ubisoft connect doesn't do this, only pointing their warnings to `D:\\JenkinsWorkspace\\workspace\\client_build_installer\\client\\products\\overlay\\mg\\overlay\\Hooks.cpp`. Either way, these paths seem to be temporal or something because I wasn't able to find them in the prefix. 

I installed Ubisoft Connect with protontricks, which in turn replaced Uplay in the wine prefix. Once I did that, launching the game from steam in turn then launched Ubisoft Connect as an updated UPlay launcher, and it was able to connect to Ubisoft services, enabling the game to run correctly. Mind you that this works with Proton Experimental!

~~Only downside (or expected behaviour?) is that the Ubisoft Connect launcher keeps running in the background even once the game is closed.~~ If this setting is changed in Ubisoft Connect, once you exit the game, steam marks the games as still running because Ubisoft Connect is running in the background. You can click on Stop in Steam to close Ubisoft Connect with this setting :)
![image](https://github.com/ValveSoftware/Proton/assets/2722773/bffb50c4-2e48-4d81-bffe-4adeb750649d)

Do I mark this issue as closed @kisak-valve ? Or maybe there's an Ubisoft launcher common issue?

EDIT: Hopefully Ubisoft/Steam updates UPlay dependant games to the latest Ubisoft Connect launcher in the future, as UPlay seems unable to connect.

**Comment by Mophead2K** [comment]
Ubisoft Connect fails to load in game mode on Steam Deck causing the entire system to crash and restart. In desktop mode this is not an issue and the game loads normally as expected.
